crypto/evp/mac_lib.c

```diff
@@ -184,3 +184,35 @@ int EVP_MAC_names_do_all(const EVP_MAC *mac,
 
     return 1;
 }
+
+unsigned char *EVP_MAC_calc(EVP_MAC *mac, const OSSL_PARAM *params,
+                            const void *key, int keylen,
+                            const unsigned char *data, size_t datalen,
+                            unsigned char *out, size_t outsize,
+                            unsigned int *outlen)
```

Comment:
This is fine, but you could *additionally* have this (the EVP_Q prefix was suggested for these kinds of "quick" functions):

``` C
unsigned char *EVP_Q_mac(OSSL_LIB_CTX *libctx, const char *mac_name, const char *propq,
                         const OSSL_PARAM *params, const void *key, int keylen,
                         const unsigned char *data, size_t datalen,
                         unsigned char *out, size_t outsize,
                         unsigned int *outlen)
```

Reply:
I added such a function as requested, which gives a nice abstraction from `EVP_MAC`.
I just think that there the name `OSSL_Q_mac` (or even just `OSSL_mac`) fits better because the result is indendendent of the `EVP_MAC` API,
so tentatively using `OSSL_Q_mac`.

Reply:
I disagree. I think the name should be `EVP_Q_mac` (or some other variant based on the `EVP_*` namespace). We have long been pushing our users to use the "high level EVP" interface. This fits fine into that, and it would be very odd indeed to have something outside of the EVP namespace.

Reply:
All right, so will change that.